src/plist.h

```diff
@@ -291,6 +291,46 @@ static inline Obj PopPlist(Obj list)
     return val;
 }
 
+
+/****************************************************************************
+**
+*F  NewEmptyPlist() . . . . . . . . . . . . . . . .  Create a new empty Plist
+*F  NewImmutableEmptyPlist() . . . . . . . Create a new immutable empty Plist
+*/
+static inline Obj NewEmptyPlist(void)
+{
+    return NEW_PLIST(T_PLIST_EMPTY, 0);
+}
+
+static inline Obj NewImmutableEmptyPlist(void)
+{
+    return NEW_PLIST_IMM(T_PLIST_EMPTY, 0);
+}
+
+/****************************************************************************
+**
+*F  NewPlistFromArray(<list>,<length>) . . . . .Create a Plist from a C array
+*/
+static inline Obj NewPlistFromArray(Obj * list, Int length)
+{
+    if(length == 0) {
+        return NewEmptyPlist();
+    }
+
+    Obj o = NEW_PLIST(T_PLIST, length);
+    SET_LEN_PLIST(o, length);
+    memcpy(BASE_PTR_PLIST(o), list, length * sizeof(Obj));
+    return o;
+}
+
+/****************************************************************************
+**
+*F  NewPlistFromArgs(<args...>) . . . Create a Plist from a list of arguments
+*/
+#define NewPlistFromArgs(...)                                                \
+    NewPlistFromArray((Obj[]){ __VA_ARGS__ },                                \
+                      (sizeof((Obj[]){ __VA_ARGS__ }) / sizeof(Obj)))
```

Comment:
This use C99 variadic macros as well as C99 compound literals, both features not everybody working on the GAP kernel may be familiar with. I think it would be helpful to add a comment above which briefly explains that?

Reply:
agreed